File: backend/app/models/bug.py
from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db.database import Base


class Bug(Base):
    __tablename__ = "bugs"

    id = Column(Integer, primary_key=True, index=True)
    project_id = Column(
        Integer,
        ForeignKey("projects.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    title = Column(String(200), nullable=False)
    severity = Column(String(50), nullable=False, default="MEDIUM")
    status = Column(String(50), nullable=False, default="OPEN")
    tech_stack = Column(String(255), nullable=True)
    error_message = Column(Text, nullable=True)
    logs = Column(Text, nullable=True)
    root_cause = Column(Text, nullable=True)
    fix_summary = Column(Text, nullable=True)
    ai_analysis = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    project = relationship("Project", back_populates="bugs")

File: frontend/src/types/bug.ts
export interface Bug {
  id: number;
  project_id: number;
  title: string;
  severity: string;
  status: string;
  tech_stack?: string | null;
  error_message?: string | null;
  logs?: string | null;
  root_cause?: string | null;
  fix_summary?: string | null;
  ai_analysis?: string | null;
  created_at: string;
  updated_at: string;
}

export interface BugCreatePayload {
  title: string;
  severity?: string;
  status?: string;
  tech_stack?: string | null;
  error_message?: string | null;
  logs?: string | null;
  root_cause?: string | null;
  fix_summary?: string | null;
  ai_analysis?: string | null;
}

export interface BugUpdatePayload {
  title?: string;
  severity?: string;
  status?: string;
  tech_stack?: string | null;
  error_message?: string | null;
  logs?: string | null;
  root_cause?: string | null;
  fix_summary?: string | null;
  ai_analysis?: string | null;
}
